[Bug]: getInstance() returns null in Vite dev although the plugin-created instance exists globally

### Describe the bug

  In Vite dev mode, `@module-federation/vite` creates a host runtime instance during federation bootstrap, but both runtime accessors return `null`:

  - `getInstance()` from `@module-federation/enhanced/runtime`
  - `getInstance()` from `@module-federation/runtime`

  The instance does exist in:

  ```ts
  globalThis.__FEDERATION__.__INSTANCES__

  but it is not returned by either getInstance() API.

### Version

7.3.2

### Reproduction

https://github.com/francesc79/mf-vite-get-instance-repro  -> pnpm i && pnpm dev

### Relevant log output

```shell
{
    "enhancedRuntimeGetInstanceReturnedNull": true,
    "enhancedRuntimeGetInstanceName": null,
    "runtimeGetInstanceReturnedNull": true,
    "runtimeGetInstanceName": null,
    "globalInstanceNames": ["__mfe_internal__host_app"]
  }

  So the plugin-created host instance exists globally, but both public runtime APIs return null.
```

### Validations

- [x] Read the [docs](https://github.com/module-federation/vite).
- [x] Read the [common issues list](https://github.com/module-federation/vite/issues).
- [x] Check that there isn't [already an issue](https://github.com/module-federation/vite/issues) that reports the same bug to avoid creating a duplicate.
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

We might be seeing a diefferent variant of the same root cause with @module-federation/vite:1.15.1 + vite:8.0.10
I've put together a [reproduction](https://github.com/robinengeldmde/mf-vite8-repro) of the issue in a minimal setup.

It surfaces as:
```
installHook.js:1 Error: [ Federation Runtime ]: Please call createInstance first. #RUNTIME-009
args: {}
View the docs to see how to solve: https://module-federation.io/guide/troubleshooting/runtime#runtime-009
    at main.ts:8:3
overrideMethod	@	installHook.js:1
(anonymous)	@	client:510
(anonymous)	@	main.ts:11
```

Sidenote: With our production codebase we also see a racecondition in Edge (Chrome & Firefox work fine) where ReactSharedInternals.H is null. It occures under the exact same conditions as this issue but only in DEV mode. Might this be related or worth a separate issue? We weren't able to do a simple reproduction of this issue thus far.

--- a/examples/vite-webpack-rspack/host/src/App.tsx
+++ b/examples/vite-webpack-rspack/host/src/App.tsx
@@ -1,3 +1,4 @@
+import { registerRemotes } from '@module-federation/runtime';
 import React, { lazy, StrictMode, Suspense, useEffect } from 'react';
 import { createRoot } from 'react-dom/client';
 import { Footer } from './components/Footer';
@@ -5,7 +6,6 @@ import { Header } from './components/Header';
 import { Toggle } from './components/Toggle';
 import { useDynamicImport } from './hooks/useDynamicImport';
 import './index.css';
-import { mfRuntime } from './mfRuntime';
 
 import _ from 'lodash';
 _.VERSION;
@@ -63,7 +63,9 @@ const TestsScreen = lazyWithRetry(
     import('testsRemote/TestsScreen')
 );
 
-mfRuntime.registerRemotes([
+// if module federation is already setup through plugin in vite.config.js,
+// there is no need to call init() from '@module-federation/runtime' here
+registerRemotes([
   {
     name: 'dynamicRemote',
     entry: 'http://localhost:4002/remoteEntry.js',

--- a/examples/vite-webpack-rspack/host/src/hooks/useDynamicImport.tsx
+++ b/examples/vite-webpack-rspack/host/src/hooks/useDynamicImport.tsx
@@ -1,5 +1,5 @@
+import { loadRemote } from '@module-federation/runtime';
 import { ElementType, useEffect, useState } from 'react';
-import { mfRuntime } from '../mfRuntime';
 
 interface DynamicImportProps {
   module: string;
@@ -12,7 +12,7 @@ async function loadRemoteWithRetry(id: string, retries = 20, delayMs = 500) {
   let lastError: unknown;
   for (let attempt = 1; attempt <= retries; attempt++) {
     try {
-      return await mfRuntime.loadRemote(id);
+      return await loadRemote(id);
     } catch (error) {
       lastError = error;
       if (attempt < retries) {

--- a/examples/vite-webpack-rspack/host/src/main.jsx
+++ b/examples/vite-webpack-rspack/host/src/main.jsx
@@ -1,3 +1,8 @@
-import './mfRuntime';
+import { init } from '@module-federation/runtime';
+
+init({
+  name: 'host',
+  remotes: [],
+});
 
 import('./App');

--- a/examples/vite-webpack-rspack/host/src/mfRuntime.js
+++ b/examples/vite-webpack-rspack/host/src/mfRuntime.js
@@ -1,6 +1,0 @@
-import { createInstance } from '@module-federation/runtime';
-
-export const mfRuntime = createInstance({
-  name: 'host',
-  remotes: [],
-});

--- a/src/virtualModules/__tests__/virtualRemoteEntry.test.ts
+++ b/src/virtualModules/__tests__/virtualRemoteEntry.test.ts
@@ -270,7 +270,7 @@ describe('virtualRemoteEntry', () => {
     );
     // Shim must appear before any imports so it's defined when component code executes
     const shimIndex = code.indexOf('__VUE_HMR_RUNTIME__');
-    const importIndex = code.indexOf('import {createInstance');
+    const importIndex = code.indexOf('import {init as runtimeInit');
     expect(shimIndex).toBeLessThan(importIndex);
   });
 

--- a/src/virtualModules/virtualRemoteEntry.ts
+++ b/src/virtualModules/virtualRemoteEntry.ts
@@ -315,7 +315,7 @@ export function generateRemoteEntry(
   if (typeof __VUE_HMR_RUNTIME__ === 'undefined') {
     globalThis.__VUE_HMR_RUNTIME__ = { createRecord() {}, rerender() {}, reload() {} };
   }
-  import {createInstance, loadRemote} from "@module-federation/runtime";
+  import {init as runtimeInit, loadRemote} from "@module-federation/runtime";
   ${pluginImportNames.map((item) => item[1]).join('\n')}
   ${
     command === 'build'
@@ -326,7 +326,6 @@ export function generateRemoteEntry(
   const initTokens = {}
   const shareScopeName = ${JSON.stringify(options.shareScope)}
   const mfName = ${JSON.stringify(options.internalName)}
-  let runtimeInstance
   let localSharedImportMapPromise
   let exposesMapPromise
   const shouldRetrySharedInitError = ${command !== 'build'} && ((error) => {
@@ -369,19 +368,13 @@ export function generateRemoteEntry(
   async function init(shared = {}, initScope = []) {
     const {usedShared, usedRemotes} = await getLocalSharedImportMap()
     ${generateDirectSharedCacheSeedCode(command)}
-    const runtimeOptions = {
+    const initRes = runtimeInit({
       name: mfName,
       remotes: usedRemotes,
       shared: usedShared,
       plugins: [${pluginImportNames.map((item) => `${item[0]}(${item[2]})`).join(', ')}],
       ${options.shareStrategy ? `shareStrategy: '${options.shareStrategy}'` : ''}
-    };
-    if (!runtimeInstance) {
-      runtimeInstance = createInstance(runtimeOptions);
-    } else {
-      runtimeInstance.initOptions(runtimeOptions);
-    }
-    const initRes = runtimeInstance;
+    });
     // handling circular init calls
     var initToken = initTokens[shareScopeName];
     if (!initToken)

